dpnp/backend/extensions/vm/types_matrix.hpp

```diff
@@ -276,6 +276,36 @@ struct DivOutputType
         dpctl_td_ns::DefaultResultEntry<void>>::result_type;
 };
 
+/**
+ * @brief A factory to define pairs of supported types for which
+ * MKL VM library provides support in oneapi::mkl::vm::exp<T> function.
+ *
+ * @tparam T Type of input vector `a` and of result vector `y`.
+ */
+template <typename T>
+struct ExpOutputType
+{
+    using value_type = typename std::disjunction<
+        dpctl_td_ns::TypeMapResultEntry<T, double, double>,
```

Comment:
result type can be omitted when the same as source type
```suggestion
        dpctl_td_ns::TypeMapResultEntry<T, double>,
```